src/conn/conn_ckpt.c

```diff
@@ -83,6 +83,14 @@ __ckpt_server(void *arg)
 
 	while (F_ISSET(conn, WT_CONN_SERVER_RUN) &&
 	    F_ISSET(conn, WT_CONN_SERVER_CHECKPOINT)) {
+		/*
+		 * Wait...
+		 * NOTE: If the user only configured logsize, then usecs
+		 * will be 0 and this wait won't return until signalled.
+		 */
+		WT_ERR(
+		    __wt_cond_wait(session, conn->ckpt_cond, conn->ckpt_usecs));
+
```

Comment:
@michaelcahill Do we want to check the flags after waking up in this case now?  On close, the flags get cleared and  the condition signaled.  This thread will wake up in that case and instead of exiting, it will perform the checkpoint.  Before it would wake up and exit.